src/go/rpk/pkg/cli/cmd/wasm/deploy.go

```diff
@@ -0,0 +1,173 @@
+package wasm
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/Shopify/sarama"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/utils"
+)
+
+func NewDeployCommand(
+	fs afero.Fs,
+	createProducer func(bool, int32) (sarama.SyncProducer, error),
+	adminCreate func() (sarama.ClusterAdmin, error),
+) *cobra.Command {
+	var description string
+
+	command := &cobra.Command{
+		Use:	"deploy <path>",
+		Short:	"deploy inline WASM function",
+		Args:	cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			path := args[0]
+			fullFileName := filepath.Base(path)
+			fileExt := filepath.Ext(fullFileName)
+			// validate file extension, just allow js extension
+			if fileExt != ".js" {
+				return fmt.Errorf("can't deploy '%s': only .js files are supported.", path)
+			}
+			fileName := strings.Replace(fullFileName, fileExt, "", 1)
+			fileContent, err := afero.ReadFile(fs, path)
+			if err != nil {
+				return err
+			}
+			// create producer
+			producer, err := createProducer(false, -1)
+			if err != nil {
+				return err
+			}
+			// create admin
+			admin, err := adminCreate()
+			if err != nil {
+				return err
+			}
+
+			return deploy(
+				fileName,
+				fileContent,
+				description,
+				producer,
+				admin,
+			)
+		},
+	}
+
+	command.Flags().StringVar(
+		&description,
+		"description",
+		"",
+		"description about what or how wasm function do, in order to "+
+			"give some information to another developers",
+	)
+
+	return command
+}
+
+/**
+this function create and publish message for deploying coprocessor
+message format:
+{
+	key: filename,
+	header: {
+		action: "deploy",
+		sha256: file_content_sha256,
+		description: file_description
+	}
+	message: binary_file_content
+}
+*/
+func deploy(
+	fileName string,
+	fileContent []byte,
+	description string,
+	producer sarama.SyncProducer,
+	admin sarama.ClusterAdmin,
+) error {
+	exist, err := kafka.ExistingTopic(admin, kafka.CoprocessorTopic)
+	if err != nil {
+		return err
+	}
+	if exist == false {
+		err := kafka.CreateCoprocessorTopic(admin)
+		return err
+	}
+	// create headers
+	headers, err := createHeaderDeployMessage("deploy", description, fileContent)
+	if err != nil {
+		return err
+	}
+	// create message
+	messageContent, err := utils.WriteBuffer([]byte{}, fileContent)
+	if err != nil {
+		return err
+	}
+	// publish message
+	error := kafka.PublishMessage(messageContent, fileName, producer, kafka.CoprocessorTopic, headers)
```

Comment:
This line is longer than 80 chars.

Reply:
Avoid naming vars the same as types. `error` is a type. It's more idiomatic to reuse a var named `err` everywhere in the function, even if you mutate it.